test_utils/src/lib.rs

```diff
@@ -313,3 +328,68 @@ pub fn resolve_name_and_committee(
 
     (name, committee)
 }
+
+////////////////////////////////////////////////////////////////
+/// Batches
+////////////////////////////////////////////////////////////////
+
+// Fixture
+pub fn batch() -> Batch {
+    Batch(vec![transaction(), transaction()])
+}
+
+// Fixture
+pub fn batch_digest() -> BatchDigest {
+    resolve_batch_digest(serialized_batch())
+}
+
+// Fixture
+pub fn resolve_batch_digest(batch_serialised: Vec<u8>) -> BatchDigest {
+    BatchDigest::new(crypto::blake2b_256(|hasher| {
+        hasher.update(&batch_serialised)
+    }))
+}
+
+// Fixture
+pub fn serialized_batch() -> Vec<u8> {
+    serialise_batch(batch())
+}
+
+pub fn serialise_batch(batch: Batch) -> Vec<u8> {
+    let message = worker::WorkerMessage::<Ed25519PublicKey>::Batch(batch);
+    bincode::serialize(&message).unwrap()
+}
+
+/// generate multiple fixture batches. The number of generated batches
+/// are dictated by the parameter num_of_batches.
+pub fn batches(num_of_batches: usize) -> Vec<Batch> {
+    let mut batches = Vec::new();
+
+    for i in 1..num_of_batches + 1 {
+        batches.push(batch_with_transactions(i));
+    }
+
+    batches
+}
+
+pub fn batch_with_transactions(num_of_transactions: usize) -> Batch {
+    let mut transactions = Vec::new();
+
+    for _ in 0..num_of_transactions {
+        transactions.push(transaction());
+    }
+
+    Batch(transactions)
+}
+
+const BATCHES_CF: &str = "batches";
+
+pub fn open_batch_store() -> Store<BatchDigest, worker::SerializedBatchMessage> {
```

Comment:
Since we are moving this , then we could also probably move the [create_db_stores](https://github.com/MystenLabs/narwhal/blob/767a364e9937d38e52db1df406aa9bfc3bb4d1eb/primary/src/tests/common.rs#L26) ? I know it's very `primary` specific, but since we are doing for the batch_store it would be nice to have all in one place.

Reply:
There's unfortunately many such functions all over the code base still, including here but also in consensus ...
The place where those leftovers are still useful is in limiting cyclic dependencies from the test utils to every other crate: we wouldn't want the testing of e.g. primary to be blocked by worker (the way it is now due to such a dependency, though it will go away when I fix #188 ).